mathesar/api/serializers/constraints.py

```diff
@@ -70,6 +72,11 @@ def create(self, validated_data):
                 raise base_api_exceptions.MathesarAPIException(e)
         return constraint
 
+    def validate_name(self, name):
+        if is_identifier_too_long(name):
+            raise database_api_exceptions.IdentifierTooLong(field='name')
+        return name
+
```

Comment:
Notice I've not added tests for constraint names. Afaict, our constraint names are automatically generated, which I presume takes the length limit into account. Still, I added this validator in case there is a bug, so that we fail early.